### PR TITLE
All harvesters load up reagents on afterattack

### PR DESCRIPTION
**Pull request by fleewave**
## About The Pull Request

All Vali weapons will now autoload after hitting a target, as the claymore currently does.
## Why It's Good For The Game

#12687 added autoloading, but only to the claymore. I don't see a reason why we can't have this on the rest of the harvester weapons as well, considering it doesn't add much of an advantage over spamming your unique-action key.
## Changelog
:cl:
qol: All harvester weapons now autoload chems after attacking.
/:cl:

### CODE REVIEW

**Comment by Xander3359** [comment]
Remove the argument instead, don't just do `TRUE` to every single instance.

**Comment by fleewave** [comment]
> Remove the argument instead, don't just do `TRUE` to every single instance.

sure, good plan

File: code/datums/components/harvester.dm [hunk 1]
```diff
@@ -30,18 +30,16 @@
 	var/datum/action/harvester/reagent_select/reagent_select_action
 	///The maximum amount that one chemical can be loaded
 	var/max_loadable_reagent_amount = 30
-	var/loadup_on_attack = FALSE
+	var/loadup_on_attack = TRUE
```
Comment (Xander3359) [inline code/datums/components/harvester.dm:33]: Just delete this entirely and bake it in to default behaviour

**Comment by Looking4bros** [comment]
> considering it doesn't add much of an advantage over spamming your unique-action key.

As if auto loading could stop me from spamming unique-action, unless it were to send ~10 "Kelotane" messages when it's loaded i will continue to do so.

File: code/datums/components/harvester.dm (hunk 1 above)
Comment (fleewave) [reply code/datums/components/harvester.dm]: done, don't know why i didn't do this in the first place

**Review by Xander3359 — APPROVED** [review]
code lgtm. 

I was considering this when adding it to Claymore but didn't really feel the need to add it to the other weapons. But I think this is probably fine

**Comment by Clay500z** [comment]
>unless it were to send ~10 "Kelotane" messages when it's loaded i will continue to do so.

So true bestie

**Comment by Looking4bros** [comment]
:eyes:

**Comment by Looking4bros** [comment]
:eye: :lips: :eye:

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**